objects/src/notes/metadata.rs

```diff
@@ -17,12 +18,13 @@ use super::{
 pub struct NoteMetadata {
     sender: AccountId,
     tag: Felt,
+    note_type: NoteType,
 }
 
 impl NoteMetadata {
     /// Returns a new [NoteMetadata] instantiated with the specified parameters.
-    pub fn new(sender: AccountId, tag: Felt) -> Self {
-        Self { sender, tag }
+    pub fn new(sender: AccountId, tag: Felt, note_type: NoteType) -> Self {
```

Comment:
Similar nit as above: I'd move `note_type` parameter before the `tag` parameter.

Reply:
moved